base/irrationals.jl

```diff
@@ -119,10 +119,10 @@ macro irrational(sym, val, def)
     qsym = esc(Expr(:quote, sym))
     bigconvert = isa(def,Symbol) ? quote
         function Base.convert(::Type{BigFloat}, ::Irrational{$qsym})
-            c = BigFloat()
+            c = Ref(BigFloat())
             ccall(($(string("mpfr_const_", def)), :libmpfr),
                   Cint, (Ref{BigFloat}, Int32), c, MPFR.ROUNDING_MODE[])
-            return c
+            return c[]
```

Comment:
The Ref is not doing anything and you **ARE** mutating the `BigFloat` so adding a `Ref` seems like a bad idea.